invenio_oauthclient/handlers/base.py

```diff
@@ -103,7 +114,48 @@ def base_authorized_signup_handler(resp, remote, *args, **kwargs):
                 db.session.commit()
                 raise OAuthClientMustRedirectSignup()
 
-        # Authenticate user
+        account_groups = handlers["groups"](resp, user)
+        if account_groups:
+            roles_id = []
+            for group in account_groups:
+                existing_role = current_datastore.find_role_by_id(group["id"])
+                if existing_role and existing_role.is_managed:
+                    current_app.logger.exception(
+                        f'Error while syncing roles: A managed role with id: ${group["id"]} already exists'
+                    )
+                    continue
+                existing_role_by_name = current_datastore.find_role(group["name"])
```

Comment:
question: why do we check if the name is existing? It does make much sense, but maybe it could be that 2 roles have different IDs, but same name? Or not?

Reply:
No, name has to be unique (here the [model](https://github.com/inveniosoftware/invenio-accounts/blob/master/invenio_accounts/models.py#L69)), therefore we check if there is an already existing managed role to avoid conflicts. I adde those checks because if there is an error it would rollback all the changes. An alternative would be commit one by one.